marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -215,36 +217,24 @@ instance semigroupHoles :: Semigroup Holes where
 instance monoidHoles :: Monoid Holes where
   mempty = Holes mempty
 
-{- | Find holes with the same name
-  We collect all the `Holes` using `getHoles` and then we fold through the result
-  to find occurances that have the same name as well as changing the values to be
-  single MarloweHoles rather than an array of MarloweHoles
--}
-validateHoles :: Holes -> Tuple (Array MarloweHole) (Map String MarloweHole)
-validateHoles (Holes m) = foldlWithIndex f (Tuple [] mempty) m
-  where
-  f k (Tuple duplicates uniquesMap) [ v ] = Tuple duplicates (Map.insert k v uniquesMap)
-
-  f k (Tuple duplicates uniquesMap) vs = Tuple (duplicates <> vs) uniquesMap
+insertHole :: forall a. IsMarloweType a => Term a -> Holes -> Holes
+insertHole (Term _ _ _) m = m
 
-insertHole :: forall a. IsMarloweType a => Holes -> Term a -> Holes
-insertHole m (Term _) = m
-
-insertHole (Holes m) (Hole name proxy start end) = Holes $ Map.alter f name m
+insertHole (Hole name proxy start end) (Holes m) = Holes $ Map.alter f name m
```

Comment:
flipping the args in `insertHole` and `getHoles` allows us to do some clever monoid tricks in places to make the code nicer.